common/print.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2023 Red Hat Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above
+ *       copyright notice, this list of conditions and the
+ *       following disclaimer.
+ *     * Redistributions in binary form must reproduce the
+ *       above copyright notice, this list of conditions and
+ *       the following disclaimer in the documentation and/or
+ *       other materials provided with the distribution.
+ *     * The names of contributors to this software may not be
+ *       used to endorse or promote products derived from this
+ *       software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ *
+ * Author: Zoltan Fridrich <zfridric@redhat.com>
+ */
+
+#include "config.h"
+
+#include "print.h"
+
+#include <stdarg.h>
+#include <unistd.h>
+
+static const char *
+style_to_sgr (enum p11_print_style style)
+{
+	switch (style) {
+	case P11_PRINT_BOLD:    return "1";
+	case P11_PRINT_BLACK:   return "30";
+	case P11_PRINT_RED:     return "31";
+	case P11_PRINT_GREEN:   return "32";
+	case P11_PRINT_YELLOW:  return "33";
+	case P11_PRINT_BLUE:    return "34";
+	case P11_PRINT_MAGENTA: return "35";
+	case P11_PRINT_CYAN:    return "36";
+	case P11_PRINT_WHITE:   return "37";
+	default:                return "0";
+	}
+}
+
+void
+p11_highlight_word (FILE *fp,
+		    const char *string)
+{
+	if (isatty (fileno (fp)))
+		fprintf (fp, "\e]8;;%s\e\\\033[36m%s\033[0m\e]8;;\e\\\n", string, string);
+	else
+		fputs (string, fp);
+}
+
+void
+p11_print_word (FILE *fp,
+		const char *string,
+		int n_styles,
+		...)
```

Comment:
Using varargs to just enumerate styles seems like a bit overkill to me. If we do this I would at least omit `n_styles` and let the caller to add a terminator, e.g., `p11_print_word (fp, string, P11_PRINT_BOLD, P11_PRINT_BLACK, P11_PRINT_INVALID)`.

Or, given only a single color would be specified, it could be turned into a flag:
```c
p11_print_word (fp, string, P11_PRINT_BLACK, P11_PRINT_BOLD|P11_PRINT_UNDERLINE);
```